### PR TITLE
Rkovalenko circle of chained words

### PR DESCRIPTION
**Pull request by rokovalenko**
I tried some basic test cases, seem to be working fine.

### CODE REVIEW

**Comment by rokovalenko** [comment]
I tried some basic test cases, seem to be working fine.

**Comment by vlemish** [comment]
will not work for smth like: 'aa', 'bb'. Should be false, but it returns true.

**Comment by PShchegolevatykh** [comment]
@vlemish good point, not everything works, e.g. cases like { "aa", "bb", "aa", "bb", "cc", "dd" }

@quadreff Also what this type of expression means?

dict.AddOrIncrementValue(word[^1]);

Maybe leave some comments around implementation.

**Comment by rokovalenko** [comment]
@PShchegolevatykh @vlemish updated the solution, it's kind of messy but should work fine.
[^1] stands for the last char in a string, .NET 6 feature I believe 😄

**Comment by rokovalenko** [comment]
@PShchegolevatykh added the second solution (IsLooped2) based on your PR, check it out

File: rkovalenko/CircleOfChainedWords.cs [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+            if(!listOfWords.Any() || listOfWords.Count == 1)
+            {
+                return false;
+            }
+
+            var startingWordIndex = 0;
+            var count = listOfWords.Count;
+            for(var i = 1; i < count;)
+            {
+                // compare last char of one word and the first char of the second word
```
Comment (PShchegolevatykh) [inline rkovalenko/CircleOfChainedWords.cs:16]: Thanks for comments

File: rkovalenko/CircleOfChainedWords.cs [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+
+            if (!listOfWords.Any() || listOfWords.Count == 1)
+            {
+                return false;
+            }
+
+            var firstWord = listOfWords[0];
+
+            //handle edge case here like a,a,a,a,a or a,b,a,a
+            if(listOfWords.All(x => x.Length == 1))
```
Comment (PShchegolevatykh) [inline rkovalenko/CircleOfChainedWords.cs:58]: Nice

File: rkovalenko/CircleOfChainedWords.cs [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+                if(listOfWords[i][^1] != listOfWords[overlappingIndex][0])
+                {
+                    return false;
+                }
+            }
+
+            return true;
+        }
+
+        public static bool IsLooped2(List<string> listOfWords)
```
Comment (PShchegolevatykh) [inline rkovalenko/CircleOfChainedWords.cs:47]: IsLooped2 still returns true for {"aa", "bb", "aa", "bb", "cc", "dd"}, not all cases work

File: rkovalenko/CircleOfChainedWords.cs [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+            var startingWordIndex = 0;
+            var count = listOfWords.Count;
+            for(var i = 1; i < count;)
+            {
+                // compare last char of one word and the first char of the second word
+                if (listOfWords[startingWordIndex][^1] == listOfWords[i][0])
+                {
+                    if(i - startingWordIndex > 1)
+                    {
+                        var overlappingIndex = (startingWordIndex + 1) % count;
```
Comment (PShchegolevatykh) [inline rkovalenko/CircleOfChainedWords.cs:21]: Index arithmetic and overlaps can benefit from some more explanatory comments ;)

File: rkovalenko/CircleOfChainedWords.cs [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+
+                    i = startingWordIndex + 1;
+                }
+                else
+                {
+                    ++i;
+                }
+            }
+
+            for(var i = 0; i < count; ++i)
```
Comment (PShchegolevatykh) [inline rkovalenko/CircleOfChainedWords.cs:35]: Not sure why the second loop to double check if we have a full chain from the first one.

**Review by PShchegolevatykh — COMMENTED** [review]
IsLooped2 does not work with all my cases, IsLooped works though so marking it as complete and merging.

**Pull request merged**